### PR TITLE
Upgrade Android and iOS SDK to 2.17.1

### PR DESCRIPTION
**Pull request by TheRusskiy**
## Summary

Upgrade Android and iOS SDK to 2.17.1.

Added wisePad3s, wisePosEDevkit and stripeS700Devkit readers to Android (as they are added in the SDK)
and wisePosEDevkit and stripeS700Devkit to iOS (also added to SDK).

Remove `connectEmbeddedReader` from Android because it was removed from the SDK. It was never present on iOS.

Everything else is satisfying new types and re-generating docs.

## Motivation

New SDK versions have fixes and it's been a while since this library synced with them.

## Testing

- [x] I tested this manually
- [ ] I added automated tests

I did not add any tests but changed existing ones.

## Documentation

Select one:

- [x] I have added relevant documentation for my changes.
- [ ] This PR does not result in any developer-facing changes.

Documentation seems to be auto-generated so I ran the command to generate it.

### CODE REVIEW

File: android/src/main/java/com/stripeterminalreactnative/ktx/TerminalExtensions.kt [hunk 1]
```diff
@@ -131,10 +121,12 @@ suspend fun Terminal.connectReader(
...
         else connectHandoffReader(reader, HandoffConnectionConfiguration(locationId))
     }
-    EMBEDDED -> connectEmbeddedReader(reader, EmbeddedConnectionConfiguration(Unit))
     USB -> {
         if (listener is UsbReaderListener)
             connectUsbReader(reader, UsbConnectionConfiguration(locationId), listener)
         else connectUsbReader(reader, UsbConnectionConfiguration(locationId))
     }
+    else -> {
+        throw IllegalArgumentException("Unsupported discovery method: ${discoveryMethod}")
```
Comment (TheRusskiy) [inline android/src/main/java/com/stripeterminalreactnative/ktx/TerminalExtensions.kt:130]: EMBEDDED is still present in types, but according to Android release notes it's not for public use

File: ios/StripeTerminalReactNative.swift [hunk 2]
```diff
@@ -312,7 +312,7 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
         let receiptEmail = params["receiptEmail"] as? String
         let customer = params["customer"] as? String
         let transferGroup = params["transferGroup"] as? String
-        let metadata = params["metadata"] as? [AnyHashable : Any]
+        let metadata = params["metadata"] as? [String : String]
```
Comment (TheRusskiy) [inline ios/StripeTerminalReactNative.swift:315]: metadata changed type:
https://github.com/stripe/stripe-terminal-ios/blob/master/CHANGELOG.md#2170-2023-01-30

File: src/hooks/useStripeTerminal.tsx [hunk 3]
```diff
@@ -396,7 +375,7 @@ export function useStripeTerminal(props?: Props) {
   );
 
   const _connectHandoffReader = useCallback(
-    async (params: ConnectEmbeddedParams) => {
+    async (params: ConnectHandoffParams) => {
```
Comment (TheRusskiy) [inline src/hooks/useStripeTerminal.tsx:378]: this was a bug in types

**Review by billfinn-stripe — APPROVED** [review]
Looks great, thanks. All of your notes made sense. 👍

**Comment by billfinn-stripe** [comment]
Bitrise wasn't able to run the integration tests [0], I think because the PR came from a fork. I pushed a branch at the same commit hash, and most of the build passed [1]. The `deploy-firebase-android`, `e2e-android`, and `e2e-ios` builds are failing. The e2e tests are also failing on 2023-02-04 for a dependabot build [2], so I'm fairly sure those are unrelated to this PR. The `deploy-firebase-android` failed with an auth error [3], and I suspect it's related to us rolling the Firebase token on 2023-01-05.

We should remove the CircleCI build configs, given that we've moved to Bitrise.

I'm going to bypass the branch protections and merge this into `main`.

[0] https://app.bitrise.io/app/4828104fe6b05c04/pipelines/37a6140d-8af9-4d76-b189-3cfeea30dcaa
[1] https://app.bitrise.io/app/4828104fe6b05c04/pipelines/47bdda48-5479-4143-a6ed-bb91eb99bc23
[2] https://app.bitrise.io/app/4828104fe6b05c04/pipelines/b38e3b13-d4a9-4b03-ac2e-c6cb3114170a
[3]
```
App Distribution could not generate credentials from the refresh token specified. For more information, try again with firebase_app_distribution's "debug" parameter set to "true".
```

**Pull request merged**